### ISSUE TITLE
ScriptBuilder AddOperations (operations []bytes)

### ISSUE THREAD

**Issue by ggeorgiev** [btcsuite/btcd#715]
I have an algorithm that computes different parts of the scripts concurrently. Right now I have to collect all the prebuilt parts and stick them together in a binary presentation on my own.

Having AddOperations public interface in ScriptBuilder will allow me to do this in a more clean way. In fact, I noticed that there is such interface implemented already, but it is in the tests. Any special reason why this is not a public interface? Ofter all using or missing it is up to the consumers of the library.

**Comment by dajohi** [comment]
where in the tests?

**Comment by davecgh** [comment]
I'm also not sure which tests @ggeorgiev is referring to.  However, based on what he said about working on scripts concurrently, I'm guessing that he wants something like:

```Go
func (b *ScriptBuilder) AddOps(opcodes []byte) *ScriptBuilder {
	if b.err != nil {
		return b
	}

	// Pushes that would cause the script to exceed the largest allowed
	// script size would result in a non-canonical script.
	if len(b.script)+len(opcodes) > maxScriptSize {
		str := fmt.Sprintf("adding opcodes would exceed the maximum "+
			"allowed canonical script length of %d", maxScriptSize)
		b.err = ErrScriptNotCanonical(str)
		return b
	}

	b.script = append(b.script, opcodes...)
	return b
}
```

**Comment by davecgh** [comment]
I went ahead and implemented this along with some tests via PR #734 since I had already written the code for the issue anyways.

**Comment by ggeorgiev** [comment]
Sorry, I did notice the questions. Dave is right - this is exactly what I need. Thanks for adding this.
